File: src/m/solvers/crpbjacobioptions.py
from pairoptions import pairoptions
from collections import OrderedDict


def crpbjacobioptions(*args):

    #retrieve options provided in *args
    options = pairoptions(*args)
    solverOptions = OrderedDict()
    solverOptions['toolkit'] = 'petsc'
    solverOptions['mat_type'] = options.getfieldvalue('mat_type', 'mpiaij')
    solverOptions['ksp_type'] = options.getfieldvalue('ksp_type', 'cr')
    solverOptions['pc_type'] = options.getfieldvalue('pc_type', 'pbjacobi')

    return solverOptions

File: src/m/solvers/crpbjacobioptions.m
function solverOptions=crpbjacobioptions(varargin)
%CRPBJACOBIOPTIONS - define PETSc solver options for the Conjugate Residual (CR) Krylov method with Point-block Jacobi preconditioning
%
%   Usage:
%      solverOptions=crpbjacobioptions(varargin);

%retrieve options provided in varargin
options=pairoptions(varargin{:});
solverOptions=struct();
solverOptions.toolkit='petsc';
solverOptions.mat_type=getfieldvalue(options, 'mat_type','mpiaij');
solverOptions.ksp_type=getfieldvalue(options, 'ksp_type','cr');
solverOptions.pc_type=getfieldvalue(options, 'pc_type',  'pbjacobi');

